bolt/lib/Rewrite/DWARFRewriter.cpp

```diff
@@ -672,22 +672,36 @@ void DWARFRewriter::updateDWARFObjectAddressRanges(
     }
   }
 
+  Optional<AttrInfo> LowPCAttrInfo =
+      findAttributeInfo(DIE, dwarf::DW_AT_low_pc);
+  Optional<AttrInfo> HighPCAttrInfo =
+      findAttributeInfo(DIE, dwarf::DW_AT_high_pc);
   if (AbbreviationDecl->findAttributeIndex(dwarf::DW_AT_ranges)) {
     // Case 1: The object was already non-contiguous and had DW_AT_ranges.
     // In this case we simply need to update the value of DW_AT_ranges
     // and introduce DW_AT_GNU_ranges_base if required.
     Optional<AttrInfo> AttrVal = findAttributeInfo(DIE, dwarf::DW_AT_ranges);
-
     std::lock_guard<std::mutex> Lock(DebugInfoPatcherMutex);
     DebugInfoPatcher.addLE32Patch(
         AttrVal->Offset, DebugRangesOffset - DebugInfoPatcher.getRangeBase(),
         AttrVal->Size);
-    if (!RangesBase)
+
+    // high_pc attribute is unused, nullify if exists
+    if (HighPCAttrInfo) {
+      if (isHighPcFormEightBytes(HighPCAttrInfo->V.getForm()))
+        DebugInfoPatcher.addLE64Patch(HighPCAttrInfo->Offset, 0);
+      else
+        DebugInfoPatcher.addLE32Patch(HighPCAttrInfo->Offset, 0);
+    }
+
+    if (!RangesBase) {
+      // low_pc attribute is unused, nullify if exists
+      if (LowPCAttrInfo)
+        DebugInfoPatcher.addLE64Patch(LowPCAttrInfo->Offset, 0);
```

Comment:
What about if it's DWARF4 and debug fission or DWARF5? It will be an index in to .debug_addr section.

Reply:
I will need to investigate this question, I'm not extremely familiar with dwarf format to be honest :) If you have the solution for this please let me know.

Reply:
Something like
```
if (LowForm == dwarf::DW_FORM_GNU_addr_index ||
      LowForm == dwarf::DW_FORM_addrx) {
   uint32_t AddressIndex =
        AddrWriter->getIndexFromAddress(0, *Unit->getDWOId());
        DebugInfoPatcher.addUDataPatch(LowPCOffset, AddressIndex,
                                    std::abs(int(HighPCOffset - LowPCOffset)));
}
```
Code assumes highpc/lowpc are next to each other.
Should handle DWARF4 Debug fission. Fine to not handle DWARF5 just add // TODO. I am still working on adding that support.

Reply:
Hello! Sorry for late response, I had a weekend :) 
Thanks for your code, but isn't it another functionality, that is should not be it handled in separate patch? As I understand this case is not handled currently?

Reply:
I don't quite remember what that tests tests for, but it might not just trigger this path.
Well as it stands now you will break DebugFission if this path is hit. Because you are always adding a 64bit patch.

Reply:
I think I understood you, what if we'll came up with the solution like this?
```
    if (!RangesBase) {
      if (LowPCAttrInfo && HighPCAttrInfo)
        convertToRangesPatchDebugInfo(DIE, DebugRangesOffset, DebugInfoPatcher);
      else if (LowPCAttrInfo)
        DebugInfoPatcher.addLE64Patch(LowPCAttrInfo->Offset, 0);
      return;
    }

```

Reply:
Not sure this will do what you want.
Why not just check for form like like I suggested, or even what you posted yesterday?
I don't think my suggestion ads a new functionality. It does same thing you are doing now: zeroing out low_pc, it just handles variable encoding used for DWARF4 Debug Fission, or DWARF5.

On side note DW_AT_high_pc and DW_AT_ranges. I am not sure that even follows the DWARF spec. When it talks about it refers either to having DW_AT_low_pc/DW_AT_high_pc or DW_AT_ranges. How is Golang uses DW_AT_high_pc, with DW_AT_ranges?

Reply:
@ayermolo If the variant I've posted yesterday is good I will endup with this one, I've probably misunderstood you. 
Speaking of highpc check - it doesn't use this, I've made extra nullify in case it exists to handle it the same way as low pc, but probably you're right and it is not needed

Reply:
ah. Then let's remove high_pc.
Sure, sounds good.